### PR TITLE
Add include logs functionality

### PR DESCRIPTION
**Pull request by chaitanyaphalak**
Set splunk.com/include annotation to true on pod and .Values.logsCollection.containers.useSplunkIncludeAnnotation flag to true to include its logs from ingested to your Splunk platform deployment. 
All other logs will be ignored.

### CODE REVIEW

**Review by dmitryax — COMMENTED** [review]
Overall looks good. 
Please delete the attribute once it's used in the `resource/logs` processor.
2. Please add the same in the otelCollector and disable it in the otelAgent in the `otelCollector.enabled=true`. The reason is that there is no k8s metadata on the agent if the gateway is enabled.

File: helm-charts/splunk-otel-collector/values.yaml [hunk 1]
```diff
@@ -332,7 +332,6 @@ logsCollection:
     # Extra operators for container logs.
     # https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/operators/README.md#what-operators-are-available
     extraOperators: []
-
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:335]: nit: unexpected removal? :)

File: helm-charts/splunk-otel-collector/templates/config/_common.tpl [hunk 2]
```diff
@@ -103,6 +105,21 @@ resource/logs:
     {{- end }}
 {{- end }}
 
+{{/*
+Include logs processor
+*/}}
+{{- define "splunk-otel-collector.includeLogsProcessors" -}}
+# If .Values.logsCollection.useSplunkIncludeAnnotation is set to true, only logs for pods with splunk.com/include annotation are ingested
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_common.tpl:112]: `.Values.logsCollection.useSplunkIncludeAnnotation` is wrong I believe. But do we need to mention the condition in the unconditional helper? Maybe just remove this line? Seems like there are enough other comments added already

File: helm-charts/splunk-otel-collector/values.yaml [hunk 3]
```diff
@@ -365,6 +365,9 @@ logsCollection:
     #       value: server
     #     firstEntryRegex: ^[^\s].*
 
+    # Set `splunk.com/include` annotation to true on pod and `logsCollection.useSplunkIncludeAnnotation` flag to `true` to include its logs from ingested to your Splunk platform deployment.
+    # All other logs will be ignored. You cant use this feature with the above mentioned exclude feature. You can only use either the include feature or the exclude feature.
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:369]: Oh missed one last thing. There is nothing "above mentioned". Please rephrase :)

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 4]
```diff
@@ -221,6 +230,9 @@ service:
         - batch
         - filter/logs
         - resource/logs
+        {{- if .Values.logsCollection.containers.useSplunkIncludeAnnotation }}
+        - filter/include_pod_logs
+        {{- end }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:235]: This filter should go before `resource/logs` to apply the include attribute removal. same in `otelCollector`

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 5]
```diff
@@ -434,6 +443,9 @@ service:
         {{- end }}
         - k8sattributes
         - batch
+        {{- if .Values.logsCollection.containers.useSplunkIncludeAnnotation }}
+        - filter/include_pod_logs
+        {{- end }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:448]: This should go under the next condition `{{- if not .Values.otelCollector.enabled }}`. Otherwise, with `otelCollector=enable`, it won't do anything because there is no k8s metatada available at this point to filter the logs on.

Another question. Do we still need to handle `splunk.com/exclude` if `logsCollection.containers.useSplunkIncludeAnnotation=true`? 

If not, we can avoid setting up unnecessary processors and reduce amount of code by just changing the helper for `filter/logs`:

```
filter/logs:
  logs:
    {{ .Values.logsCollection.containers.useSplunkIncludeAnnotation | "include" "exclude" }}:
      match_type: strict
      resource_attributes:
        - key: splunk.com/{{ .Values.logsCollection.containers.useSplunkIncludeAnnotation | "include" "exclude" }}
          value: "true"
```

What do you think?

Comment (chaitanyaphalak) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:448]: really good idea, i think we should just use your definition of filter/logs

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 6]
```diff
@@ -282,6 +282,11 @@ processors:
         - key: splunk.com/index
           tag_name: com.splunk.index
           from: pod
+        {{- if .Values.logsCollection.containers.useSplunkIncludeAnnotation }}
+        - key: splunk.com/include
+          tag_name: splunk.com/include
+          from: pod
+        {{- end }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:289]: Can we take `splunk.com/exclude` annotation under `else` clause to avoid unnecessary attributes manipulations?

File: helm-charts/splunk-otel-collector/values.yaml [hunk 7]
```diff
@@ -365,6 +365,9 @@ logsCollection:
     #       value: server
     #     firstEntryRegex: ^[^\s].*
 
+    # Set `splunk.com/include` annotation to true on pod and `logsCollection.useSplunkIncludeAnnotation` flag to `true` to include its logs from ingested to your Splunk platform deployment.
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:368]:
```suggestion
    # Set `splunk.com/include` annotation to true on pod and `logsCollection.containers.useSplunkIncludeAnnotation` flag to `true` to include its logs from ingested to your Splunk platform deployment.
```

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl (hunk 6 above)
Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:289]: Btw why don't we take `splunk.com/include` from namespace? Just curious

Comment (chaitanyaphalak) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:289]: The feature request from feedback was only for pods. We could add namespace support in future if there is customer demand!

Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:289]: Do you see any issue with having consistent behavior for the filtering from the start?

If you agree, we could even add a helper for the filtering attribute that would return `splunk.com/include` or `splunk.com/exclude` depending on the configuration. Then all the places where `splunk.com/exclude` is used can be just replaced with `{{ include "splunk-otel-collector.filterAttr" . }}`. It'll remove unnecessary attribute manipulations which is always good for performance

Comment (chaitanyaphalak) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:289]: I think we should probably do that later during the refactor.

Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:289]: This doesn't seem to be a big change, not sure why it has to introduce any tech debt

File: helm-charts/splunk-otel-collector/templates/config/_common.tpl [hunk 8]
```diff
@@ -153,12 +155,28 @@ Filter logs processor
...
 {{- end }}
 
+{{/*
+Filter Attributes Function
+*/}}
+{{- define "splunk-otel-collector.filterAttr" -}}
+{{- if .Values.logsCollection.containers.useSplunkIncludeAnnotation }}
+- key: splunk.com/include
+  tag_name: splunk.com/include
+  from: pod
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_common.tpl:172]: My idea was to return only `splunk.com/include` or `splunk.com/exclude` in this helper and update _all_ the places where `splunk.com/exclude` currently is being used:
- resource/logs delete action
- k8sattributes processor
- filter/logs processor

For example
```yaml
resource/logs:
  attributes:
    - key: {{ include "splunk-otel-collector.filterAttr" . }}
       action: delete
```

WDYT? Can we make it this way? And here I'm assuming we apply it consistently to pods and namespaces. I still don't understand why we have to introduce this inconsistency just because a customer wants it to be scraped from pods for now. I believe having it as a consistent config for both pods and namespaces is what can be expected by others and will not introduce any problems to the customer who asked for this feature initially.

**Comment by chaitanyaphalak** [comment]
Hey Dmitrii, updated the PR, with your latest recommendations, lint-test is failing due to some fluentd error, I am not sure why?

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 9]
```diff
@@ -56,15 +56,13 @@ processors:
...
-        - key: splunk.com/exclude
-          tag_name: splunk.com/exclude
-          from: pod
         - key: splunk.com/index
           tag_name: com.splunk.index
           from: namespace
         - key: splunk.com/index
           tag_name: com.splunk.index
           from: pod
+        {{- include "splunk-otel-collector.filterAttr" . | nindent 8 }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:65]: Please update usage here

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 10]
```diff
@@ -270,18 +270,19 @@ processors:
...
+        - key: {{ include "splunk-otel-collector.filterAttr" . }}
+          tag_name: {{ include "splunk-otel-collector.filterAttr" . }}
           from: pod
         - key: splunk.com/index
           tag_name: com.splunk.index
           from: namespace
         - key: splunk.com/index
           tag_name: com.splunk.index
           from: pod
+        {{- include "splunk-otel-collector.filterAttr" . | nindent 8 }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:285]: this is not needed

File: rendered/manifests/agent-only/configmap-otel-agent.yaml [hunk 11]
```diff
@@ -16,236 +16,4 @@ metadata:
...
-          - k8sattributes
-          - batch
-          - resource
-          - resourcedetection
-          receivers:
-          - otlp
-          - jaeger
-          - smartagent/signalfx-forwarder
-          - zipkin
+    Error: 'error converting YAML to JSON: yaml: line 123: could not find expected '':'''
```
Comment (dmitryax) [inline rendered/manifests/agent-only/configmap-otel-agent.yaml:19]: This doesn't seem to be desirabe

**Review by dmitryax — APPROVED** [review]
LGTM. did you run `make render`?

File: helm-charts/splunk-otel-collector/templates/config/_common.tpl [hunk 12]
```diff
@@ -45,6 +45,17 @@ zipkin:
 {{- end }}
 {{- end }}
 
+{{/*
+Filter Attributes Function
+*/}}
+{{- define "splunk-otel-collector.filterAttr" -}}
+{{- if .Values.logsCollection.containers.useSplunkIncludeAnnotation }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_common.tpl:52]: I think the problem is because this helper returns `\nsplunk.com/include` or `\nsplunk.com/exclude`. Please trim the space after `if` and `else`

Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/config/_common.tpl:52]:
```suggestion
{{- if .Values.logsCollection.containers.useSplunkIncludeAnnotation -}}
```

Comment (chaitanyaphalak) [reply helm-charts/splunk-otel-collector/templates/config/_common.tpl:52]: Let me try, thanks for the input!

File: helm-charts/splunk-otel-collector/values.yaml [hunk 13]
```diff
@@ -371,7 +371,9 @@ logsCollection:
     #     containerName:
     #       value: server
     #     firstEntryRegex: ^[^\s].*
-
+    # Set `splunk.com/include` annotation to true on pod and `logsCollection.containers.useSplunkIncludeAnnotation` flag to `true` to include its logs from ingested to your Splunk platform deployment.
+    # All other logs will be ignored.
+    useSplunkIncludeAnnotation: false
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:376]: Looks good to me. Just please add a line breaks above and below this section to avoid mixing with other comments

File: helm-charts/splunk-otel-collector/values.yaml [hunk 14]
```diff
@@ -371,7 +371,9 @@ logsCollection:
     #     containerName:
     #       value: server
     #     firstEntryRegex: ^[^\s].*
-
+    # Set `splunk.com/include` annotation to true on pod and `logsCollection.containers.useSplunkIncludeAnnotation` flag to `true` to include its logs from ingested to your Splunk platform deployment.
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:374]: It's not only about Splunk Platform. How about this wording?
```suggestion
    # Set this flag to `true` to collect logs from pods with `splunk.com/include: true` annotation and ignore others.
```

File: helm-charts/splunk-otel-collector/values.yaml [hunk 15]
```diff
@@ -349,6 +349,7 @@ logsCollection:
     # identify source of the logs: namespace, pod and container names. At least one source
     # identifier has to be specified in for each multiline config.
     multilineConfigs: []
+
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/values.yaml:352]: This is a wrong place to separate the comments :)

Comments below are still applied to multiline config and mixed with comments for `useSplunkIncludeAnnotation`

**Review by dmitryax — APPROVED** [review]
LGTM
